File: Unigram/Unigram.WebView/TelegramWebviewProxy.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Windows.Data.Json;
using Windows.Foundation.Metadata;

namespace Unigram.Webview
{
    [AllowForWeb]
    public sealed class TelegramWebviewProxy
    {
        private TelegramWebviewProxyDelegate _callback;

        public TelegramWebviewProxy(TelegramWebviewProxyDelegate callback)
        {
            _callback = callback;
        }

        public void PostEvent(string eventName, string eventData)
        {
            if (eventName.Equals("payment_form_submit"))
            {
                try
                {
                    var json = JsonObject.Parse(eventData);
                    var response = json.GetNamedValue("credentials");
                    var title = json.GetNamedString("title", string.Empty);

                    _callback.Invoke(title, response.Stringify());
                }
                catch
                {
                    _callback.Invoke(null, eventData);
                }
            }
        }
    }

    public delegate void TelegramWebviewProxyDelegate(string title, string credentials);
}

File: Unigram/Unigram.Native/TelegramWebviewProxy.cpp
#include "pch.h"
#include "TelegramWebviewProxy.h"
#if __has_include("TelegramWebviewProxy.g.cpp")
#include "TelegramWebviewProxy.g.cpp"
#endif

#include <winrt/Windows.Data.Json.h>

using namespace winrt::Windows::Data::Json;

namespace winrt::Unigram::Native::implementation
{
	TelegramWebviewProxy::TelegramWebviewProxy(TelegramWebviewProxyDelegate delegate)
	{
		_delegate = delegate;
	}

	void TelegramWebviewProxy::PostEvent(hstring eventName, hstring eventData)
	{
		_delegate(eventName, eventData);
		return;

		if (eventName == L"payment_form_submit")
		{
			try
			{
				auto json = JsonObject::Parse(eventData);
				auto response = json.GetNamedValue(L"credentials");
				auto title = json.GetNamedString(L"title", L"");

				_delegate(title, response.Stringify());
			}
			catch (winrt::hresult_error const& ex)
			{
				_delegate(L"", eventData);
			}
		}
	}
} // namespace winrt::Unigram::Native::implementation